commit 6dbef74aeb090d6bee7d64ef3fa82ae6fa53f271
Author: Cristian Ciocaltea <cristian.ciocaltea@collabora.com>
Date:   Fri Aug 11 23:37:05 2023 +0300

    x86/cpu/amd: Enable Zenbleed fix for AMD Custom APU 0405
    
    Commit
    
      522b1d69219d ("x86/cpu/amd: Add a Zenbleed fix")
    
    provided a fix for the Zen2 VZEROUPPER data corruption bug affecting
    a range of CPU models, but the AMD Custom APU 0405 found on SteamDeck
    was not listed, although it is clearly affected by the vulnerability.
    
    Add this CPU variant to the Zenbleed erratum list, in order to
    unconditionally enable the fallback fix until a proper microcode update
    is available.
    
    Fixes: 522b1d69219d ("x86/cpu/amd: Add a Zenbleed fix")
    Signed-off-by: Cristian Ciocaltea <cristian.ciocaltea@collabora.com>
    Signed-off-by: Borislav Petkov (AMD) <bp@alien8.de>
    Cc: stable@vger.kernel.org
    Link: https://lore.kernel.org/r/20230811203705.1699914-1-cristian.ciocaltea@collabora.com

diff --git a/arch/x86/kernel/cpu/amd.c b/arch/x86/kernel/cpu/amd.c
index 26ad7ca423e7..c15b4f02b4cf 100644
--- a/arch/x86/kernel/cpu/amd.c
+++ b/arch/x86/kernel/cpu/amd.c
@@ -73,6 +73,7 @@ static const int amd_erratum_1054[] =
 static const int amd_zenbleed[] =
 	AMD_LEGACY_ERRATUM(AMD_MODEL_RANGE(0x17, 0x30, 0x0, 0x4f, 0xf),
 			   AMD_MODEL_RANGE(0x17, 0x60, 0x0, 0x7f, 0xf),
+			   AMD_MODEL_RANGE(0x17, 0x90, 0x0, 0x91, 0xf),
 			   AMD_MODEL_RANGE(0x17, 0xa0, 0x0, 0xaf, 0xf));
 
 static bool cpu_has_amd_erratum(struct cpuinfo_x86 *cpu, const int *erratum)